client/src/pages/OnBoardingPage.tsx

```diff
@@ -1,9 +1,120 @@
-import React from 'react'
+import React, { useRef, useState } from "react";
+import styles from "./OnBoardingPage.module.css";
+import { IconLeetCodeLogo } from "../icons/icon-leetcode-logo";
+import { IconVJudgeLogo } from "../icons/icon-vjudge-logo";
+import { IconAddPhoto } from "../icons/import-photo-icon";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
 
 const OnBoardingPage = () => {
+  const hiddenFileInput = useRef<HTMLInputElement>(null);
+  const [imageUrl, setImageUrl] = useState<string | null>(null);
+  const [leetCodeUsername, setLeetCodeUsername] = useState("");
+  const [vJudgeUsername, setVJudgeUsername] = useState("");
+  const [errorMsg, setErrorMsg] = useState("");
+  const navigate = useNavigate();
+
+  const handleClick = () => {
+    hiddenFileInput.current?.click();
+  };
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const fileUploaded = event.target.files?.[0];
+    if (fileUploaded) {
+      console.log(fileUploaded);
+      setImageUrl(URL.createObjectURL(fileUploaded));
+    }
+  };
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (leetCodeUsername === "" && vJudgeUsername === "") {
+      setErrorMsg("Please ensure at least one of the usernames is provided");
+      return;
+    }
+
+    const payload = {
+      leetCodeUsername,
+      vJudgeUsername,
+      imageUrl,
+    }
+
+    setErrorMsg("");
+
+    try {
+      const response = await axios.post("http://localhost:8080/api/auth/login", payload);
```

Comment:
login requires {username, password}, one can't be logged in without those fields

Reply:
yeaaa so i didnt know where to sent this post req to

Reply:
this is just a template call i added but it technically should be the enpoint for the onboarding or something, so i was thinking on register, the user will have its account created but not finalized, and as long as they don't have a non-taken username for one of the Vjudge or leetcode, they cant continue with the application.

Reply:
yes perciely thats what i meant by " as long as they don't have a non-taken username for one of the Vjudge or leetcode, they cant continue with the application." so checking the mongo db for that